### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by goosetherumfoodle**
- added devise. Only non-destructive actions should be accessible to non-logged in users. 
- I didn't mess with the devise config files since it didn't seem necessary for just protecting pages. 
- I added a sign-out link to the layout/application view for convenience.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,8 +2,10 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
-
+  before_action :authenticate_user!, except: [:index, :show, :hello]
+  
   def hello
+
```
Comment (justinxreese) [inline app/controllers/application_controller.rb]: Kill this whitespace

File: config/environments/development.rb [hunk 2]
```diff
@@ -38,4 +38,7 @@
 
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
+
+  # Define default URL options for Devise gem
+  config.action_mailer.default_url_options = {   host: 'localhost', port: 3000 }
```
Comment (justinxreese) [inline config/environments/development.rb]: Kill the extra spaces before the 'host' key

**Comment by justinxreese** [comment]
Code looks good @goosetherumfoodle. Just left some small comments about some missed white space things.

**Comment by justinxreese** [comment]
@builderworksco/qa this is good for QA'ing

**Comment by TGizzi** [comment]
- [x] added devise. Only non-destructive actions should be accessible to non-logged in users.
- [x] I didn't mess with the devise config files since it didn't seem necessary for just protecting pages.
- [x] I added a sign-out link to the layout/application view for convenience.

Log in and out is working as intended.

**Pull request merged**